freedb/protocols.md

```diff
@@ -1,8 +1,311 @@
+
 # FreeDB Protocols
 
-We are going to explain all the details required to implement `FreeDB` in other programming languages.
-The goal is to allow people to contribute `FreeDB` implementation in other languages.
+## Row Store
+
+Row Store is a data structure that stores records as a sequence of rows (similar to RDBMS like MySQL, PostgreSQL, etc).
+
+### Data Structure
+
+The underlying format that we're going to use to store the data in Google Sheets is fairly simple. The sheet will have
+`N + 1` columns where `N` is the number of columns that the client specified in the schema. The extra 1 column will be
+used to store `_rid` metadata (which will be explained later).
+
+The first row will be used to store the column name of each column. The first column will be the `_rid` metadata column
+(which also has the `_rid` column name) and will be followed by the first column in the data schema, then followed by the
+second one, and so on. The next rows (second row and rows below that) will be used to store the records.
+
+Suppose the client schema looks like this (written in `PyFreeDB` model):
+```
+class Person(models.Model):
+    name = models.StringField() # 1st column
+    age = models.IntegerField() # 2nd column
+```
+
+and we have data below:
+```
+[
+    Person(name="A", age=1), # 1st row
+    Person(name="B", age=2), # 2nd row
+    Person(name="C", age=3), # 3rd row
+    Person(name="D", age=4)  # 4th row
+]
+```
+
+Then the actual data that we store in Google Sheet would look like this:
+| _rid | name | age |
+| ---- | ---- | --- |
+| 1    | A    | 1   |
+| 2    | B    | 2   |
+| 3    | C    | 3   |
+| 4    | D    | 4   |
+
+#### About `_rid` Metadata
+
+`_rid` column will be used to tell the index of a row. This metadata will make our Google Sheet Formula simpler
+so that we can utilize [Google Visualization API][GVizAPI] for our needs.
+
+⚠️ Even though `_rid` can be used as the row identifier (because `_rid` is unique between each row), we recommend the
+client to implement their own row identifier mechanism such as [UUID](https://en.wikipedia.org/wiki/Universally_unique_identifier)
+or [Snowflake](https://en.wikipedia.org/wiki/Snowflake_ID) because we will not guarantee that `_rid` will not be
+reused out-of-the-box (to ensure the `_rid` is not reused, you need to implement soft-delete mechanism by your own).
+
+Consider this scenario:
+1. Client inserts Row A to the store, which will have `_rid=2`.
+2. Client deletes Row A from the store.
+3. Client Insert Row B to the store, which will reuse the same `_rid` as A i.e. B's `_rid` will be equal to `2`.
+
+We did not expose this field by default because if the client didn't aware of this behavior it can cause various bugs
```

Comment:
`If the client is not aware of ...` (grammar)

Reply:
Updated

Reply:
I put this info on the top of the page (below data structure).

Instead of explaining the obvious (how to replace the optional section in the query), I replaced this section with an example instead.